### PR TITLE
fp write for STIX Objects

### PR DESCRIPTION
**Pull request by emmanvg**
Adding this piece of code would help write more flexible code while also providing the same options as ``serialize()``

### CODE REVIEW

**Review by chisholm — COMMENTED** [review]
I like this idea.  Could the `serialize()` function even be reimplemented to call `fp_serialize()` and pass an `io.StringIO`?  That would remove the code duplication between the two functions.

**Comment by emmanvg** [comment]
I started with having both under a same function, but I was doubtful about having two different behaviors (one that returned a string and one that returned None). Based on the same idea that `json.dumps()` and `json.dump()` are essentially the same except for the required `fp` argument and what each return, I thought it would be better to also keep them separate on `stix2` too.

File: stix2/serialization.py [hunk 1]
```diff
@@ -85,6 +85,44 @@ def sort_by(element):
         return json.dumps(obj, cls=STIXJSONEncoder, **kwargs)
 
 
+def fp_serialize(obj, fp, pretty=False, include_optional_defaults=False, **kwargs):
+    """
+    Serialize a STIX object as a stream to file-like supporting object.
+
+    Args:
+        obj: The STIX object to be serialized.
+        fp: A ``.write()``-supporting file-like object.
```
[the remark this answers is not in the document]
Comment (emmanvg) [reply stix2/serialization.py:94]: Good point

**Comment by chisholm** [comment]
I didn't mean to combine them into one function.  It was more like change the serialize() function to be something like:

```python
with io.StringIO() as fp:
    fp_serialize(obj, fp)
    return fp.getvalue()
```

So the "main" code for serializing lives in one place, the two functions can't get out of sync, the serialize() implementation is shorter and simpler, etc.  Just a thought.

**Comment by emmanvg** [comment]
I misinterpreted your prior message. It is a good suggestion...

**Review by clenk — APPROVED** [review]

**Pull request merged**